astropy/nddata/nduncertainty.py

```diff
@@ -358,7 +359,252 @@ def _propagate_divide(self, other_uncert, result_data, correlation):
+        a variance.
+
+        Parameters
+        ----------
+
+        other_uncert : `~astropy.nddata.NDUncertainty` instance
+            The uncertainty, if any, of the other operand.
+
+        result_data : `~astropy.nddata.NDData` instance
+            The results of the operation on the data.
+
+        correlation : float or `numpy.ndarray`-like
+            Correlation of the uncertainties.
+
+        divide : bool, optional
+            If ``True``, propagate for division, otherwise propagate for
+            multiplication.
+
+        to_variance : function, optional
+            Function that will transform the input uncertainties to variance.
+            The default assumes the uncertainty is the variance.
+
+        from_variance : function, optional
+            Function that will convert from variance to the input uncertainty.
+            The default assumes the uncertainty is the variance.
+        """
+        # For multiplication we don't need the result as quantity
+        if isinstance(result_data, Quantity):
+            result_data = result_data.value
+
+        if divide:
+            correlation_sign = -1
+        else:
+            correlation_sign = 1
+
+        if self.array is None:
+            # We want the result to have a unit consistent with the parent, so
+            # we only need to convert the unit of the other uncertainty if it
+            # is different from its data's unit.
+            if (other_uncert.unit and
+                to_variance(1 * other_uncert.unit) !=
+                    ((1 * other_uncert.parent_nddata.unit)**2).unit):
+                other = to_variance((other_uncert.array * other_uncert.unit)).to(
+                    (1 * other_uncert.parent_nddata.unit)**2).value
+            else:
+                other = to_variance(other_uncert.array)
+            if divide:
+                # Formula: sigma**2 = (A / B)**2 * (dB / B**2)
+                return from_variance(result_data**2 * other /
+                                     other_uncert.parent_nddata.data**2)
+            else:
+                # Formula: sigma**2 = |A|**2 * dB
+                return from_variance(np.abs(self.parent_nddata.data**2 * other))
+
+        elif other_uncert.array is None:
+            # Just the reversed case
+            if (self.unit and
+                to_variance(1 * self.unit) !=
+                    ((1 * self.parent_nddata.unit)**2).unit):
+                this = to_variance(self.array * self.unit).to(
+                    (1 * self.parent_nddata.unit)**2).value
+            else:
+                this = to_variance(self.array)
+            if divide:
+                # Formula: sigma**2 = dA / B**2
+                return from_variance(this / other_uncert.parent_nddata.data**2)
+            else:
+                # Formula: sigma**2 = dA * |B|**2
+                return from_variance(np.abs(other_uncert.parent_nddata.data**2 * this))
+        else:
+            # Multiplication
```

Comment:
Please check error propagation here (assuming input error is variable and that both operands have uncertainty)